astropy/io/misc/yaml.py

```diff
@@ -110,6 +110,9 @@ def _timedelta_constructor(loader, node):
 
 
 def _ndarray_representer(dumper, obj):
+    if obj.dtype.hasobject:
+        raise TypeError(f"Cannot serialize Numpy object array: {obj}")
```

Comment:
Many standard library exception messages start with lower case so I think that is the preference for new astropy code.
```
NameError: name 'variable_name' is not defined
ValueError: invalid literal for int() with base 10: 'abc'
IndexError: list index out of range
ZeroDivisionError: division by zero
```
```suggestion
        raise TypeError(f"cannot serialize numpy object array: {obj}")
```